dev/NumberBox/NumberBoxParser.h

```diff
@@ -0,0 +1,78 @@
+﻿#pragma once
+#include <vector>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <cmath>
+#include <algorithm>
+#include <iterator>
+#include <cstdlib>
+#include <regex>
+#include <stack>
+#include <optional>
+
+// Represents a token in a mathematic expression. This can be a simple operator (+-*/^), a decimal number, or parentheses. 
+class MathToken
+{
+public:
+    enum TokenType
+    {
+        Invalid = 0,
+        Numeric = 1,
+        Operator = 2,
+        ParenLeft = 3,
+        ParenRight = 4,
+        EOFToken = 6
+    };
+    TokenType type;
+    std::wstring str;
+    MathToken();
+    MathToken(TokenType t, std::wstring s);
```

Comment:
One constructor from MathToken(TokenType t, const std::wstring_view& s) should be sufficient. The member field can stay as wstring since the [small string optimization should take effect for most of these](https://shaharmike.com/cpp/std-string/).